### PR TITLE
Snapshot and Fake Tests Later

### PR DESCRIPTION
**Pull request by johguentner**
With the help of the `PestHttpRecorder::class` and `HttpRecorder::class`, all external requests (done with the `Http` Facade) during testing, will be intercepted, saved as snapshot and faked for the same requests for future tests.

This is increases the validity of all snapshots, which fake requests during testing (important for fast tests and testing within CI).

If new tests are written, the snapshots will we generated locally automatically. After breaking changes of the Notion API, the generation of the snapshots can be forced, to keep validity of tests at a high level.

### CODE REVIEW

File: tests/Pest.php [hunk 1]
```diff
@@ -1,5 +1,14 @@
 <?php
 
+use Dotenv\Dotenv;
 use FiveamCode\LaravelNotionApi\Tests\NotionApiTest;
+use Illuminate\Support\Facades\Config;
 
-uses(NotionApiTest::class)->in(__DIR__);
+// uses(NotionApiTest::class)->in(__DIR__);
```
Comment (mechelon) [inline tests/Pest.php:7]: Remove comment line

File: src/Macros/PestHttpRecorder.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        $method = Str::lower($request->method());
+        $name = Str::slug(Str::replace('/', '-', $urlInfo['path']));
+        $query = Str::slug(Str::replace('&', '_', Str::replace('=', '-', $urlInfo['query'])));
+
+        $fileName = "{$method}_{$name}_{$query}.json";
+        $directoryPath = "tests/{$this->snapshotDirectory}";
+        $filePath = "{$directoryPath}/{$fileName}";
+
+        if ($forceRecording || ! File::exists($filePath)) {
+            File::makeDirectory($directoryPath, 0777, true, true);
```
Comment (mechelon) [inline src/Macros/PestHttpRecorder.php:70]: Check file permissions - is 777 necessary?

**Review by mechelon — APPROVED** [review]

File: src/Macros/PestHttpRecorder.php (hunk 2 above)
Comment (johguentner) [reply src/Macros/PestHttpRecorder.php]: thank you for noticing! changed it now :)

File: tests/Pest.php (hunk 1 above)
Comment (johguentner) [reply tests/Pest.php]: done :)

**Pull request merged**